flink-streaming-connectors/flink-connector-rethinkdb/src/main/java/org/apache/flink/streaming/connectors/rethinkdb/FlinkRethinkDbSink.java

```diff
@@ -0,0 +1,289 @@
+	 * @param hostport
+	 * @param database
+	 * @param table
+	 * @param schema
+	 */
+	public FlinkRethinkDbSink(String hostname, int hostport, String database, String table, 
+			JSONSerializationSchema<OUT> schema) {
+		this(hostname, hostport, database, table, schema, ConflictStrategy.update);
+	}
+
+	/**
+	 * Constructor for sink
+	 * @param hostname
+	 * @param hostport
+	 * @param database name
+	 * @param table name
+	 * @param schema serialization converter
+	 * @param conflict resolution strategy for document id conflict
+	 */
+	public FlinkRethinkDbSink(String hostname, int hostport, String database, String table, 
+			JSONSerializationSchema<OUT> schema, 
+			ConflictStrategy conflict) {
+		this.hostname = Objects.requireNonNull(hostname);
+		this.hostport = hostport;
+		this.databaseName = Objects.requireNonNull(database);
+		this.tableName = Objects.requireNonNull(table);
+		this.serializationSchema = Objects.requireNonNull(schema);
+		this.conflict = conflict;
+	}
+
+	/**
+	 * Open the sink
+	 */
+	@Override
+	public void open(Configuration parameters) throws Exception {
+		LOG.info("Received parameters : {}", parameters);
+		
+		super.open(parameters);
+
+		rethinkDbConnection = getRethinkDB().connection().hostname(hostname)
+				.port(hostport).user(username, password).connect();
+
+		LOG.info("RethinkDb connection created for host {} port {} and db {}", 
+				hostname, hostport,databaseName);
+	}
+
+	/**
+	 * Helper method to help testability
+	 * @return RethinkDB instance
+	 */ 
+	protected RethinkDB getRethinkDB() {
+		return RethinkDB.r;
+	}
+	
+	/**
+	 * Set username and password. If username and password are not provided,
+	 * then default username (admin) and blank password are used.
+	 * 
+	 * @param username
+	 * @param password
+	 * 
+	 * @throws IllegalArgumentException if arguments is null or empty
+	 */
+	public void setUsernameAndPassword(String username, String password) {
+		
+		if ( StringUtils.isBlank(username) )  {
+			throw new IllegalArgumentException("username " + username + " cannot be null or empty" ); 
+		} else {
+			this.username = username;
+		}
+		
+		if ( StringUtils.isBlank(password) ) {
+			throw new IllegalArgumentException("password " + password + " cannot be null or empty" ); 
+		} else {
+			this.password = password;
+		}
+	}
+	
+	/**
+	 * Invoke the sink with the input
+	 * 
+	 * @param the value to be inserted
+	 * 
+	 * @throws RuntimeException if there are errors while inserting row into rethinkdb
+	 */
+	@Override
+	public void invoke(OUT value) throws Exception {
+		LOG.debug("Received value {}", value);
+		
+		Object json = serializationSchema.toJSON(value);
+		LOG.debug("Object/Json: {}/{}", value, json);
+		Insert insert = getRdbTable().insert(json).optArg(CONFLICT_OPT, conflict.toString());
+		HashMap<String,Object> result = runInsert(insert);
+		
+		LOG.debug("Object/Json/Result: {}/{}/{}", value, json, result);
+		
+		if ( (Long)result.get(RESULT_ERROR_KEY) != 0 ) {
```

Comment:
this is a synchronous operation, correct? If so I'd be curious about a benchmark for this sink.